### PR TITLE
[LibrariesIoTask] move getting of project url to F8AConfiguration

### PR DESCRIPTION
**Pull request by jpopelka**
Just small refactoring.

### CODE REVIEW

File: f8a_worker/defaults.py [hunk 1]
```diff
@@ -155,5 +155,26 @@ def select_random_github_token(cls):
...
 
+    @classmethod
+    def libraries_io_project_url(cls, ecosystem, name):
+        """Construct url to endpoint, which gets information about a project and it's versions."""
+        url = '{api}/{platform}/{name}'. \
+            format(api=cls.LIBRARIES_IO_API,
+                   platform=ecosystem,
+                   name=name)
+
+        if not cls.LIBRARIES_IO_TOKEN or cls.LIBRARIES_IO_TOKEN == 'not-set':
```
Comment (pkajaba) [inline f8a_worker/defaults.py]: What is the difference between `not-set` and and `no-token`?

File: f8a_worker/defaults.py [hunk 2]
```diff
@@ -155,5 +155,26 @@ def select_random_github_token(cls):
...
+            raise F8AConfigurationException("LIBRARIES_IO_TOKEN has not been set.")
+
+        # 'no-token' value forces the API call to not use ANY token.
+        # It works, but if abused, they can ban your IP, so use with caution.
+        if cls.LIBRARIES_IO_TOKEN != 'no-token':
```
Comment (pkajaba) [inline f8a_worker/defaults.py]: Is this condition really required? I believe that `not-set` and `no-token` are same, so this is redundant.

File: f8a_worker/workers/libraries_io.py [hunk 3]
```diff
@@ -45,7 +29,8 @@ def execute(self, arguments):
         if ecosystem == 'go':
             name = quote(name, safe='')
 
-        project = get_response(self.project_url(ecosystem, name))
+        project_url = self.configuration.libraries_io_project_url(ecosystem, name)
```
Comment (pkajaba) [inline f8a_worker/workers/libraries_io.py]: Why did you move `libraries_io_project_url` to configuration module? It makes more sense to place it inside `utils` for me.

File: f8a_worker/defaults.py (hunk 1 above)
Comment (jpopelka) [reply f8a_worker/defaults.py]: `not-set` is default value, see
https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/defaults.py#L58
https://github.com/fabric8-analytics/fabric8-analytics-deployment/blob/master/openshift/env-template.sh#L33
It means that user has not set the `LIBRARIES_IO_TOKEN` and we want to explicitly tell the user that it is required to do so.

`no-token` is special value. By setting `LIBRARIES_IO_TOKEN` to `no-token` the user requests that (s)he explicitly does not want to use ANY token, see the comment in code below. We use this in tests where we don't want to store the token anywhere, but still be able to run (and pass) the tests locally.
I understand that `not-set` and `no-token` sounds similar, so if you have any suggestions, how to call it, fire.
Maybe `unauthenticated` ?

File: f8a_worker/defaults.py (hunk 2 above)
Comment (jpopelka) [reply f8a_worker/defaults.py]: Yes, it is. See above comment.

File: f8a_worker/workers/libraries_io.py (hunk 3 above)
Comment (jpopelka) [reply f8a_worker/workers/libraries_io.py]: Because it's basically a property of `F8AConfiguration`. I'd mark it `@property` if it had not arguments (ecosystem, name).

File: f8a_worker/defaults.py (hunk 1 above)
Comment (pkajaba) [reply f8a_worker/defaults.py]: No, it's OK. I just wanted to be sure whether it's not mistaken. Thanks!

File: f8a_worker/workers/libraries_io.py (hunk 3 above)
Comment (pkajaba) [reply f8a_worker/workers/libraries_io.py]: Ok, thanks for an answer.
